src/peergos/server/util/HttpUtil.java

```diff
@@ -54,4 +55,131 @@ public static void replyError(HttpExchange exchange, Throwable t) {
             Logging.LOG().log(Level.WARNING, e.getMessage(), e);
         }
     }
+
+    public static byte[] get(PresignedUrl url) throws IOException {
+        try {
+            HttpURLConnection conn = (HttpURLConnection) new URI(url.base).toURL().openConnection();
+            conn.setRequestMethod("GET");
+            for (Map.Entry<String, String> e : url.fields.entrySet()) {
```

Comment:
add a timeout?

Reply:
We need a timeout with ipfs, because it can take arbitrarily long. S3 is always fast.

Reply:
But sure. Done.